Backpack-SwiftUI/SponsoredBanner/Classes/BPKSponsoredBanner.swift

```diff
@@ -0,0 +1,201 @@
+                .foregroundColor(.sponsoredBannerBackgroundColor)
+        )
+        .clipShape(
+            RoundedRectangle(cornerRadius: .sm, style: .continuous)
+        )
+        .if(customAccessibilityLabel != nil) { view in
+            view
+                .accessibilityElement(children: .ignore)
+                .accessibilityLabel(customAccessibilityLabel ?? "")
+        }
+    }
+    
+    private var bodyTextView: some View {
+        Group {
+            if hasBody && showTerms == true {
+                BPKText(bodyText ?? "", style: .caption)
+                    .lineLimit(4)
+                    .frame(maxWidth: .infinity)
+                    .padding(.base)
+                    .background(Color.clear)
+                    .zIndex(0)
+            }
+        }
+    }
+    
+    private var hasBody: Bool {
+        return bodyText != nil
+    }
+    
+    private var topView: some View {
+        Button(action: toggleBodyText) {
+            buttonContent
+        }
+        .accessibilityRemoveTraits(hasBody ? [] : .isButton)
+        .accessibilityElement(children: .combine)
+        .buttonStyle(SponsoredBannerButtonStyle(
+            foregroundColor: (variant == .onDark) ? Color(BPKColor.textOnDarkColor) : Color(BPKColor.textOnLightColor),
+            backgroundColor: backgroundColor
+        ))
+    }
+    
+    private var buttonContent: some View {
+        HStack(spacing: .md) {
+            if let logo = logo {
+                logo
+                    .resizable()
+                    .scaledToFit()
+                    .frame(maxHeight: 22)
+            }
+            titlesView
+            Spacer()
+            if let callToAction = callToAction {
+                BPKText(callToAction.text, style: .caption)
+                    .foregroundColor(variant.color)
+            }
+            if callToAction?.showIcon == true {
+                iconView
+            }
+        }
+        .padding(.base)
+    }
+    
+    private var titlesView: some View {
+        VStack(alignment: .leading, spacing: .sm) {
+            if let title = title {
+                BPKText(title, style: .label2)
+                    .foregroundColor(variant.color)
+            }
+            if let subheadline = subheadline {
+                BPKText(subheadline, style: .caption)
+                    .foregroundColor(variant.color)
+            }
+        }
+    }
+    
+    private var iconView: some View {
+        BPKIconView(.informationCircle)
+            .foregroundColor(variant.color)
+            .if(callToAction?.accessibilityLabel != nil) { icon in
+                icon
+                    .accessibilityElement()
+                    .accessibilityHint(callToAction?.accessibilityLabel ?? "")
+            }
+    }
+    
+    private func toggleBodyText() {
+        withAnimation(.spring()) {
+            showTerms.toggle()
+        }
+    }
+}
+
+extension BPKSponsoredBanner {
```

Comment:
could you move this extension to a new file? nothing bad to have it here really, just to make the files smaller and easier to read later